packages/frontend/src/features/sqlRunner/hooks/useGithubDbtWritePreview.tsx

```diff
@@ -0,0 +1,50 @@
+import {
+    type ApiError,
+    type ApiGithubDbtWritePreview,
+    type VizColumn,
+} from '@lightdash/common';
+import { useMutation } from '@tanstack/react-query';
+import { lightdashApi } from '../../../api';
+import useToaster from '../../../hooks/toaster/useToaster';
+
+type CreatePrParams = {
+    projectUuid: string;
+    name: string;
+    sql: string;
+    columns: VizColumn[];
+};
+
+const createPullRequest = async ({
+    projectUuid,
+    name,
+    sql,
+    columns,
+}: CreatePrParams) =>
+    lightdashApi<ApiGithubDbtWritePreview['results']>({
+        url: `/projects/${projectUuid}/sqlRunner/preview`,
+        method: 'POST',
+        body: JSON.stringify({ name, sql, columns }),
```

Comment:
do we need `sql` and `columns`?

Reply:
no, but since we already have `CreatePrParams` I decided to use the same type for the body, we might use it in the future. Happy to remove it if you don't like it